File: codegen/results/v002/m3-max-48gb/ollama/gemma4_31b-mlx_nothink/20260910_094114/python/001_csv_to_json/solution.py
import csv
import json
import sys

def solve():
    input_file = 'input/data.csv'
    data_list = []

    try:
        with open(input_file, mode='r', encoding='utf-8') as f:
            reader = csv.DictReader(f)
            for row in reader:
                # Convert types as specified
                item = {
                    "Name": str(row["Name"]),
                    "Age": int(row["Age"]),
                    "Email": str(row["Email"]),
                    "Score": float(row["Score"])
                }
                data_list.append(item)
        
        # Output valid JSON array to stdout
        sys.stdout.write(json.dumps(data_list, indent=2))
    except FileNotFoundError:
        pass

if __name__ == "__main__":
    solve()

File: codegen/results/v002/m3-max-48gb/ollama/gemma4_31b-mlx_nothink/20260910_094114/go/001_csv_to_json/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"fmt"
	"os"
	"strconv"
)

// User represents the structure of the JSON objects
type User struct {
	Name  string  `json:"Name"`
	Age   int     `json:"Age"`
	Email string  `json:"Email"`
	Score float64 `json:"Score"`
}

func main() {
	// Open the input file
	file, err := os.Open("input/data.csv")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error opening file: %v\n", err)
		os.Exit(1)
	}
	defer file.Close()

	// Read CSV data
	reader := csv.NewReader(file)
	records, err := reader.ReadAll()
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error reading CSV: %v\n", err)
		os.Exit(1)
	}

	if len(records) == 0 {
		fmt.Println("[]")
		return
	}

	var users []User

	// Skip header (index 0) and iterate through rows
	for i := 1; i < len(records); i++ {
		row := records[i]
		if len(row) < 4 {
			continue
		}

		// Parse Age
		age, err := strconv.Atoi(row[1])
		if err != nil {
			age = 0
		}

		// Parse Score
		score, err := strconv.ParseFloat(row[3], 64)
		if err != nil {
			score = 0.0
		}

		users = append(users, User{
			Name:  row[0],
			Age:   age,
			Email: row[2],
			Score: score,
		})
	}

	// Marshal the slice to JSON
	jsonData, err := json.MarshalIndent(users, "", "  ")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error marshaling JSON: %v\n", err)
		os.Exit(1)
	}

	// Output the JSON array to stdout
	fmt.Println(string(jsonData))
}